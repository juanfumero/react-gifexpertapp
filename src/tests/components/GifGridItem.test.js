import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Mi titulo';
    const url = 'http://localhost:3000';

    let wrapper = shallow( <GifGridItem title={ title } url={ url } />);

    beforeEach( () => {

        wrapper = shallow( <GifGridItem title={ title } url={ url } />);

    })
 
    test('debe de mostrar <GifGridItem /> correctamente', () => {

         expect( wrapper ).toMatchSnapshot();
 
    });

    test('debe de tener un pÃ¡rrafo con el title', () => {

         const p = wrapper.find('p');
         expect( p.text().trim() ).toBe(title);
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
   });

   test('debe de tener animate__animated', () => {

        const div = wrapper.find('div');
        const className = div.props().className;
        expect( className.includes('animate__animated') ).toBe(true);

    });

 });