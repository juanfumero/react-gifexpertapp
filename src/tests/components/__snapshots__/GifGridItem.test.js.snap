// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Mi titulo"
    src="http://localhost:3000"
  />
  <p>
    Mi titulo
  </p>
</div>
`;
